Source/Particles/PhotonParticleContainer.cpp

```diff
@@ -99,17 +115,15 @@ PhotonParticleContainer::Evolve (int lev,
                                        cEx, cEy, cEz,
                                        cBx, cBy, cBz,
                                        t, dt);
-
 }
 
 
 #ifdef WARPX_QED
-void PhotonParticleContainer::InitOpticalDepth(
-    WarpXParIter& pti,
-    warpx_breit_wheeler_engine& engine)
+void PhotonParticleContainer::InitOpticalDepth()
 {
+    auto pti = WarpXParIter{*this, 0};
```

Comment:
This looks wrong to me - I think you need to iterate over all the tiles in the particle container here:

```
#ifdef WARPX_QED
void PhotonParticleContainer::InitOpticalDepth()
{
    int num_levels = finestLevel() + 1;
    for (int lev = 0; lev < num_levels; ++lev)
    {
        for (WarpXParIter pti(*this, lev); pti.isValid(); ++pti)
        {
            auto& taus = pti.GetAttribs(particle_comps["tau"]);
            for(auto& tau: taus)
                tau = bw_engine->get_optical_depth();
        }
    }
}
#endif

```

Reply:
Thanks!

Reply:
I've just pushed the change